static_frame/core/loc_map.py

```diff
@@ -41,6 +41,13 @@
 _ZERO_PAD_ARRAY.flags.writeable = False
 
 
+class FirstDuplicatePosition(KeyError):
+    __slots__ = ('first_dup',)
```

Comment:
While I appreciate the sentiment, I am not sure this `__slots__` definition unless `KeyError` and all of its parents are slotted.